packages/path_provider/path_provider_android/test/path_provider_android_test.dart

```diff
@@ -89,13 +89,15 @@ void main() {
       });
     } // end of for-loop
 
-    test('getDownloadsPath fails', () async {
-      try {
-        await pathProvider.getDownloadsPath();
-        fail('should throw UnsupportedError');
-      } catch (e) {
-        expect(e, isUnsupportedError);
-      }
+    test('getDownloadsPath succeeds', () async {
+      final String? path = await pathProvider.getDownloadsPath();
+      expect(path, kExternalStoragePaths);
+    });
+
+    //TODO: override externalStoragePaths as empty
+    test('getDownloadsPath null', () async {
+      final String? path = await pathProvider.getDownloadsPath();
+      expect(path, null);
     });
```

Comment:
can check this one if its going in the right direction? I'm trying to add a case for the null externalStoragePaths, but if I override it to empty list then it will affect other tests

Reply:
You'll either have to modify `kExternalStoragePaths` and set it back or override `_Api` or `TestPathProviderApi` with the desired behavior and set your test to use that implementation like [here](https://github.com/flutter/packages/blob/2931de43755e98618e11d329ac2208dd85319857/packages/path_provider/path_provider_android/test/path_provider_android_test.dart#L48).

Reply:
I've updated the null test

Reply:
This approach seems okay to me but perhaps you can move this test to a different group and set up the `TestPathProviderApi` with `TestPathProviderApi.setup(_ApiNull())` just for that group of tests, so that it doesn't conflict with any tests in this group?

Reply:
Sorry, I had an answer to the original question that I didn't realize I'd never actually published, and was still `Pending`. It would be much simpler to make the test implementation return configurable values, rather than have two different implementations and test groups.

Reply:
I believe Stuart was suggesting making `_Api` return a configurable value (i.e. modifying `kExternalStoragePaths`) versus mocking `TestPathProviderApi`, but please correct me if I'm wrong.

Reply:
Yes, I was suggesting making the values returned by `_Api` configurable rather than hard-coded constants. Usually we do that by making the return values public fields of the fake implementation.

I'm okay with `mockito` instead (although I think it's overkill here), but not with having two different ways of structuring these tests; if you really want to use `mockito` then all of the tests need to be rewritten to use `mockito` and `_Api` removed.